client/src/models/businessConstants.ts

```diff
@@ -0,0 +1,739 @@
+import _ from "lodash";
+
+import { assign_to_dev_helper_namespace } from "src/core/assign_to_dev_helper_namespace";
+import { lang } from "src/core/injected_build_constants.ts";
+
+interface BusinessConceptTextValue {
+  text: string;
+}
+type BilingualBusinessConceptTextValue = LangDict<string>;
+
+type RawBusinessConceptInputTextValue =
+  | BilingualBusinessConceptTextValue
+  | BusinessConceptTextValue;
+
+function getNormalizedEntry(
+  obj: RawBusinessConceptInputTextValue
+): BusinessConceptTextValue {
+  if (obj.hasOwnProperty("text")) {
+    return { text: (obj as BusinessConceptTextValue).text };
+  } else if (obj.hasOwnProperty(lang)) {
+    return { text: (obj as BilingualBusinessConceptTextValue)[lang] };
+  } else {
+    throw Error("shouldn't happen");
+  }
+}
+
+interface RawBusinessConceptEntrySet {
+  [key: string]: RawBusinessConceptInputTextValue;
+}
+interface BusinessConceptEntrySet {
+  [key: string]: BusinessConceptTextValue;
+}
+
+function normalizeBilinguals(
+  entries: RawBusinessConceptEntrySet
+): BusinessConceptEntrySet {
+  return _.mapValues(entries, (entry) => getNormalizedEntry(entry));
+}
+
+type StdObjEntry = { text: string; so_num: number };
+
+export const sos = _.mapValues<
+  { [key: number]: LangDict<string> },
+  { [key: number]: StdObjEntry }
+>(
+  {
+    1: {
+      en: "Personnel",
+      fr: "Personnel",
+    },
+    2: {
+      en: "Transportation and Telecommunications",
+      fr: "Transports et communications",
+    },
+    3: {
+      en: "Information",
+      fr: "Information",
+    },
+    4: {
+      en: "Professional and Special Services",
+      fr: "Services professionnels et spéciaux",
+    },
+    5: {
+      en: "Rentals",
+      fr: "Location",
+    },
+    6: {
+      en: "Purchased Repair and Maintenance",
+      fr: "Achat de services de réparation et d'entretien",
+    },
+    7: {
+      en: "Utilities, Materials and Supplies",
+      fr: "Services publics, fournitures et approvisionnements",
+    },
+    8: {
+      en: "Acquisition of Land, Buildings, and Works",
+      fr: "Acquisition de terrains, de bâtiments et d'ouvrages",
+    },
+    9: {
+      en: "Acquisition of Machinery and Equipment",
+      fr: "Acquisition de machines et de matériel",
+    },
+    10: {
+      en: "Transfer Payments",
+      fr: "Paiements de transfert",
+    },
+    11: {
+      en: "Public Debt Charges",
+      fr: "Frais de la dette",
+    },
+    12: {
+      en: "Other Subsidies and Payments",
+      fr: "Autres subventions et paiements",
+    },
+    20: {
+      en: "Revenues",
+      fr: "Revenus",
+    },
+    21: {
+      en: "External Revenues",
+      fr: "Revenus externes",
+    },
+    22: {
+      en: "Internal Revenues",
+      fr: "Revenus internes",
+    },
+  },
+  (val: BilingualBusinessConceptTextValue, key) => ({
```

Comment:
Just curious, seeing some places where you declare (some) types for arguments to lodash predicates and other places where you don't. Got a pattern for that or is it something you're playing by ear?

Reply:
Just fiddling with it until TS stops complaining. The first time I went through all the business constants, I wasn't aware that mapValues had generics, so I didn't use them. I'm not entirely sure in which cases we'll need them or not, but they were useful the 2nd time